### PR TITLE
feat: Grouped messages based on date and added time for the messages

### LINKED ISSUE #242: Time at which message was sent.
1>In Messages add a message stating at what time the message was sent as marked in below screenshot.
2>Also group the messages on basis of day like yesterday, 2 days ago , 4 days ago for this you can make use of this package https://www.npmjs.com/package/react-time-ago

In the messages model we have createdAt field stating at what time the message was created. You can make use of it and implement the above features

Happy Coding 💯

Screenshot (598)

### PR DESCRIPTION
**Pull request by balaraju-nerati**
I have added time for the messages and Organized the messages based on date.

**Screenshot of the feature** 
Screenshot 2024-06-09 200035

Closes #242

### CODE REVIEW

**Comment by SamarthKadam** [comment]
@balaraju-nerati great work. But make a little change that box appears a liltle big decrease its font a little bit and also decrease the padding as it looks big
Screenshot (606)

**Comment by balaraju-nerati** [comment]
@SamarthKadam sure! I'll update it

**Comment by balaraju-nerati** [comment]
Hey, @SamarthKadam , I modified the code regarding the changes u mentioned.

**Comment by SamarthKadam** [comment]
@balaraju-nerati  why am I not able to see timestamps in some messages

**Comment by SamarthKadam** [comment]
@balaraju-nerati  why am I not able to see timestamps in some messages
Screenshot (607)

**Comment by balaraju-nerati** [comment]
Hey, @SamarthKadam sry for that bug. Didn't see that. Now it's been solved. Please check now and let me know if there are any issues.

**Comment by Spiral-Memory** [comment]
Hey @SamarthKadam 
Could you please approve my request for Vercel deployment preview? After that I'll review the PR and test the changes. 

image

**Comment by SamarthKadam** [comment]
@Spiral-Memory now is it visible?

**Comment by Spiral-Memory** [comment]
> @Spiral-Memory now is it visible?

Nope

**Comment by Spiral-Memory** [comment]
Yes, I've started the review and have suggested some changes.. Waiting for the contributor to reply !

**Comment by balaraju-nerati** [comment]
Hey, @Spiral-Memory I don't know what changes you requested to me regarding this Pull request. Can you please explain?

**Comment by Spiral-Memory** [comment]
> Hey, @Spiral-Memory I don't know what changes you requested to me regarding this Pull request. Can you please explain?
> 

Sorry ! Haven't submitted the review.. check now !

**Comment by SamarthKadam** [comment]
@Spiral-Memory @souravbhunia07 done reviewing. As this pr was made 5 days ago. Please make this fast

**Comment by Spiral-Memory** [comment]
> @Spiral-Memory @souravbhunia07 done reviewing. As this pr was made 5 days ago. Please make this fast

I have finished reviewing @SamarthKadam . I am waiting for the contributor to make the suggested changes. Aren't my review comments available?

**Comment by Spiral-Memory** [comment]
If my review comments are not available for any reason, i am attaching the screenshot of the same.

image
image
image

**Comment by Spiral-Memory** [comment]
@balaraju-nerati Please make the required changes.

**Comment by balaraju-nerati** [comment]
Hey, @Spiral-Memory I'm working on the changes you suggested. I'll update by the end of tomorrow. Sry about the delay.

**Comment by balaraju-nerati** [comment]
Hey, @Spiral-Memory Please check now....I have optimized the code as per your suggestions.

**Comment by Spiral-Memory** [comment]
Hey @balaraju-nerati 

Once completed and is ready for review. do let me know..

**Comment by Spiral-Memory** [comment]
> Hey, @Spiral-Memory Please check now....I have optimized the code as per your suggestions.
>  

Cool sure ! 
Will review it tomorrow morning

File: frontend/src/pages/Search.js
Comment (Spiral-Memory) [inline frontend/src/pages/Search.js:76]: Remove all console logs please !

Comment (SamarthKadam) [reply frontend/src/pages/Search.js:76]: @balaraju-nerati Yeah remove all the console.logs present in frontend

**Comment by Spiral-Memory** [comment]
LGTM @balaraju-nerati, I am approving this PR, please make sure to remove the console logs after that it can be merged. Thanks!

**Review by Spiral-Memory — APPROVED** [review]

File: frontend/src/pages/Search.js
Comment (balaraju-nerati) [reply frontend/src/pages/Search.js]: Hey, @SamarthKadam I've removed the console logs.

Comment (Spiral-Memory) [reply frontend/src/pages/Search.js]: LGTM

**Pull request merged**
